### PR TITLE
🚸 Display usageInfo based on input type

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-04-22 下午4 32 58
截圖 2023-04-22 下午4 33 15

### CODE REVIEW

File: pages/view/_iscnId.vue [hunk 1]
```diff
@@ -303,15 +303,7 @@
...
-            <Link
-              :class="[
-                'text-[14px]',
-                'break-all',
-              ]"
-              :href="metadata.usageInfo"
-            >
-              {{ metadata.usageInfo }}
-            </Link>
+            {{ metadata.usageInfo }}
```
Comment (williamchong) [inline pages/view/_iscnId.vue:306]: usageInfo can be a url, we should check if it is an url and use link tag, otherwise this patch makes sense

**Review by williamchong — CHANGES_REQUESTED** [review]

**Comment by ckxpress** [comment]
does it mean i can put "https://creativecommons.org/licenses/by-nc-sa/4.0/" in usageInfo field in iscn.json and have it correctly displayed?

**Review by williamchong — APPROVED** [review]
Probably a simple startsWith() is sufficient though

**Review by nwingt — APPROVED** [review]

File: pages/view/_iscnId.vue [hunk 2]
```diff
@@ -304,6 +304,7 @@
             class="mb-[12px]"
           >
             <Link
+              v-if="urlPattern.test(metadata.usageInfo)"
```
Comment (nwingt) [inline pages/view/_iscnId.vue:307]: make this as a computed prop, then you don't have to inject `URL_REGEX` into data

File: pages/view/_iscnId.vue [hunk 3]
```diff
@@ -657,6 +661,14 @@ export default class ViewIscnIdPage extends Vue {
     return '';
   }
 
+  get isValidUrl() {
```
Comment (williamchong) [inline pages/view/_iscnId.vue:664]:
```suggestion
  get isUsageInfoUrl() {
```

File: pages/view/_iscnId.vue [hunk 4]
```diff
@@ -304,6 +304,7 @@
             class="mb-[12px]"
           >
             <Link
+              v-if="isValidUrl"
```
Comment (williamchong) [inline pages/view/_iscnId.vue:307]:
```suggestion
              v-if="isUsageInfoUrl"
```

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/view/_iscnId.vue [hunk 5]
```diff
@@ -657,6 +661,14 @@ export default class ViewIscnIdPage extends Vue {
     return '';
   }
 
+  get isValidUrl() {
+    return (
+      this.metadata.usageInfo.startsWith('http://') ||
+      this.metadata.usageInfo.startsWith('https://') ||
+      this.metadata.usageInfo.startsWith('ftp://')
```
Comment (williamchong) [inline pages/view/_iscnId.vue:668]: Should check if `this.metadata` and `this.metadata.usageInfo` exists
i.e
```
      this.metadata?.usageInfo && (
      this.metadata.usageInfo.startsWith('http://') ||
      this.metadata.usageInfo.startsWith('https://') ||
      this.metadata.usageInfo.startsWith('ftp://'))
```
Not sure if this is valid syntax, please try

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**